r4/src/main/java/org/opencds/cqf/r4/providers/MeasureOperationsProvider.java

```diff
@@ -221,86 +223,156 @@ public MeasureReport evaluateMeasure(@IdParam IdType theId,
+        if (status == null || status.isEmpty()) {
+            throw new IllegalArgumentException("Status is required.");
         }
-        if(null != practitioner && null == organization){
-            throw new IllegalArgumentException("If a practitioner is specified then an organization must also be specified.");
+        for (String statusValue: status) {
+            if(!Strings.isNullOrEmpty(statusValue) && (!statusValue.equalsIgnoreCase("open-gap") && !statusValue.equalsIgnoreCase("closed-gap"))){
+                throw new IllegalArgumentException("Status must be either 'open-gap', 'closed-gap', or both.");
+            }
         }
-        if(null == subject && null == practitioner && null == organization){
-            throw new IllegalArgumentException("periodStart AND periodEnd AND (subject OR organization OR (practitioner AND organization)) MUST be provided");
+        if (measureIdentifier != null && !measureIdentifier.isEmpty()) {
+            throw new NotYetImplementedException("measureIdentifier Not Yet Implemented.");
         }
         return true;
     }
 
-    private Bundle patientCareGap(String periodStart, String periodEnd, String subject, String topic, String measure, String status) {
+    private List<Measure> resolveMeasures(List<String> measureId, List<String> measureIdentifier, List<CanonicalType> measureUrl) {
+        List<Measure> measures = new ArrayList<Measure>();
+        if (measureId == null || measureId.isEmpty()) {
+            logger.info("No measure Ids found.");
+        } else {
+            measureId.forEach(id -> {
+                Measure measure = this.measureResourceProvider.getDao().read(new IdType(id));
+                if (measure != null) {
+                    measures.add(measure);
+                }
+            });
+        }
+        logger.info("measureIdentifier: " + measureIdentifier + "Not Yet Supported");
+        if (measureUrl == null || measureUrl.isEmpty()) {
+            logger.info("No measure urls found.");
+        } else {
+            measureUrl.forEach(url -> {
+                Measure measure = resolveMeasureByUrl(url);
+                if (measure != null) {
+                    measures.add(measure);
+                }
+            });
+        }
+        return measures;
+    }
+
+    private Measure resolveMeasureByUrl(CanonicalType url) {
+        String urlValue = url.getValueAsString();
+        if (!urlValue.contains("/Measure/")) {
+            throw new IllegalArgumentException("Invalid resource type for determining Measure from url: " + url);
+        }
+        String [] urlsplit = urlValue.split("/Measure/");
+        if (urlsplit.length != 2) {
+            throw new IllegalArgumentException("Invalid url, Measure.url SHALL be <CanonicalBase>/Measure/<MeasureName>");
+        }
+        String canonicalBase = urlsplit[0];
+   
+        String measureName = urlsplit[1];
+        IdType measureIdType = new IdType();
```

Comment:
Here you actually need to do a lookup based on Measure name in order to get the Id.

Reply:
Added a TODO as this is a WIP (to be worked on at the ConneConnectathon).

Reply:
Talked with Rob out of band, going to work on this in the upcoming connectathon